tests/hive-new-test.c

```diff
@@ -0,0 +1,126 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <CUnit/Basic.h>
+#if defined(_WIN32) || defined(_WIN64)
+#include <windows.h>
+#include <Shellapi.h>
+#endif
+#if defined(HAVE_UNISTD_H)
+#include <unistd.h>
+#endif
+#include <hive_impl.h>
+
+static hive_1drv_opt_t onedrv_option;
+
+int onedrv_open_oauth_url(const char *url)
+{
+#if defined(_WIN32) || defined(_WIN64)
+   ShellExecute(NULL, "open", url, NULL, NULL, SW_SHOWNORMAL);
+#else
+    char open_url[PATH_MAX];
+    memcpy(open_url, "open ", 5);
+    memcpy(open_url+5, url, strlen(url)+1 );
+
+    system(open_url);
+#endif
+    return 0;
+}
+
+static void test_hive_init(void)
+{
+    int rc;
+
+    rc = hive_global_init();
+    CU_ASSERT_EQUAL_FATAL(rc, 0);
+
+    hive_global_cleanup();
+    return;
+}
+
+static void test_hive_double_init(void)
+{
+    int rc;
+
+    rc = hive_global_init();
+    CU_ASSERT_EQUAL_FATAL(rc, 0);
+
+    rc = hive_global_init();
+    CU_ASSERT_EQUAL_FATAL(rc, 0);
+
+    hive_global_cleanup();
+    return;
+}
+
+static void test_hive_new(void)
+{
+    int rc;
+    hive_t *hive;
+
+    rc = hive_global_init();
+    CU_ASSERT_EQUAL_FATAL(rc, 0);
+
+    hive = hive_new((hive_opt_t *)(&onedrv_option));
+    CU_ASSERT_PTR_NOT_NULL_FATAL(hive);
+
+    hive_kill(hive);
+    hive = NULL;
+	
+    onedrv_option.base.type = HIVE_TYPE_NONEXIST;
+
+    hive = hive_new((hive_opt_t *)(&onedrv_option));
+    CU_ASSERT_PTR_NULL_FATAL(hive);
+
+    hive_kill(hive);
+    hive_global_cleanup();
+
+    return;
+}
+
+static void test_hive_new_without_init(void)
```

Comment:
Test cases should be independent. Here, test_hive_new_without_init() depends on running test_hive_new().